### PR TITLE
[WIP] Add models

### PR DESCRIPTION
**Pull request by jimxlin**
As of this PR I'm failing tests for the Game scope.

### CODE REVIEW

File: .tags [hunk 1]
```diff
@@ -0,0 +1,31 @@
+!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
```
Comment (jacobwgillespie) [inline .tags:1]: This file shouldn't be committed - you can remove it with `git rm .tags` and then add `.tags` to the `.gitignore` file and commit that.

File: app/models/game.rb [hunk 2]
```diff
@@ -0,0 +1,10 @@
+class Game < ApplicationRecord
+  has_many :participations
+
+  scope :available,
+    # -> { where('(white_player_id = ? AND black_player_id != ?)
+    #             OR
+    #             (white_player_id != ? AND black_player_id = ?)',
+    #             nil, nil, nil, nil) }
+    -> { where('(white_player_id = ? OR black_player_id = ?)', nil, nil) }
```
Comment (jacobwgillespie) [inline app/models/game.rb:9]: So, two things - first it seems like Postgres is having an issue with the `= nil` syntax and actually wants you to use `IS NULL` and `IS NOT NULL`.  That fixes the first test failure, however the one that omits games with two missing players then fails.  For that you want to find games missing one player or the other but not both.  Something like:

```ruby
class Game < ApplicationRecord
  has_many :participations

  scope :available, -> { 
    where('(white_player_id IS NULL AND black_player_id IS NOT NULL) OR (white_player_id IS NOT NULL AND black_player_id IS NULL)')
  }
end
```

**Comment by jimxlin** [comment]
Thanks! All tests now pass.

File: .tags (hunk 1 above)
Comment (jimxlin) [reply .tags:1]: Got it.

**Pull request closed without merging**